client/src/rubin/nublado/client/testing/_jupyter.py

```diff
@@ -296,41 +304,49 @@ def missing_lab(self, request: Request) -> Response:
         user = self._get_user_from_headers(request)
         if user is None:
             return Response(403, request=request)
+        if redirect := self._maybe_redirect(request, user):
+            return Response(
+                302, request=request, headers={"Location": redirect}
+            )
         assert str(request.url).endswith(f"/hub/user/{user}/lab")
         return Response(503, request=request)
 
     def lab(self, request: Request) -> Response:
         user = self._get_user_from_headers(request)
         if user is None:
             return Response(403, request=request)
+        if redirect := self._maybe_redirect(request, user):
+            return Response(
+                302, request=request, headers={"Location": redirect}
+            )
         assert str(request.url).endswith(f"/user/{user}/lab")
         if JupyterAction.LAB in self._fail.get(user, {}):
             return Response(500, request=request)
         state = self.state.get(user, JupyterState.LOGGED_OUT)
         if state == JupyterState.LAB_RUNNING:
             # In real life, there's another redirect to
-            # /hub/api/oauth2/authorize, which doesn't set a cookie,
-            # and then redirects to /user/username/oauth_callback.
+            # /hub/api/oauth2/authorize, which doesn't set a cookie, and then
+            # redirects to /user/username/oauth_callback.
             #
-            # We're skipping that one since it doesn't change the
-            # client state at all.
+            # We're skipping that one since it doesn't change the client state
+            # at all.
             xsrf = f"_xsrf={self._lab_xsrf}"
             return Response(
                 302,
                 request=request,
                 headers={
-                    "Location": _url(
-                        self._base_url, f"user/{user}/oauth_callback"
-                    ),
+                    "Location": self._url(f"user/{user}/oauth_callback"),
                     "Set-Cookie": xsrf,
                 },
             )
         else:
+            if self._use_subdomains:
+                host = "nb." + self._base_url.host
```

Comment:
Is `nb.` actually constant?  I guess effectively it is, whether or not it comes from something settable in values.yaml.

Reply:
Yeah, for the test mock it is constant, similar to how the test mock hard-codes `/nb`. I don't think we would gain anything from making that configurable in tests. I might be missing something, though, and if so, let me know!